cmod/board.py

```diff
@@ -1,194 +1,297 @@
-      self.det_map[det].vis_coord = format_dict(
-          jsonmap[det]['Visual coordinates'])
-      self.det_map[det].vis_M = format_dict(jsonmap[det]['FOV transformation'])
-
-  def save_calib_file(self, file):
-    dicttemp = {det: self.det_map[det].__dict__() for det in self.det_map}
-
-    with open(file, 'w') as f:
-      f.write(json.dumps(dicttemp, indent=2))
-
-  def get_det(self, detid):
-    return self.det_map[detid]
-
-  def dets(self):
-    return self.det_map.keys()
-
-  def calib_dets(self):
-    return sorted([k for k in self.dets() if int(k) < 0], reverse=True)
-
-  def add_calib_det(self, detid, mode=-1, channel=-1):
-    detid = str(detid)
-    if detid not in self.dets() and int(detid) < 0:
-      self.det_map[detid] = Detector({
-          "mode": mode,
-          "channel": channel,
-          "default coordinates": [-100, -100]
-      }, self)
-
-  # Get/Set calibration measures with additional parsing
-  def add_vis_coord(self, det, z, data):
-    det = str(det)
-    self.det_map[det].vis_coord[self.roundz(z)] = data
-
-  def add_visM(self, det, z, data):
-    det = str(det)
-    self.det_map[det].vis_M[self.roundz(z)] = data
-
-  def add_lumi_coord(self, det, z, data):
-    det = str(det)
-    self.det_map[det].lumi_coord[self.roundz(z)] = data
-
-  def get_vis_coord(self, det, z):
-    det = str(det)
-    return self.det_map[det].vis_coord[self.roundz(z)]
-
-  def get_visM(self, det, z):
-    det = str(det)
-    return self.det_map[det].vis_M[self.roundz(z)]
-
-  def get_lumi_coord(self, det, z):
-    det = str(det)
-    return self.det_map[det].lumi_coord[self.roundz(z)]
-
-  def vis_coord_hasz(self, det, z):
-    det = str(det)
-    return self.roundz(z) in self.det_map[det].vis_coord
-
-  def visM_hasz(self, det, z):
-    det = str(det)
-    return self.roundz(z) in self.det_map[det].vis_M
-
-  def lumi_coord_hasz(self, det, z):
-    det = str(det)
-    return self.roundz(z) in self.det_map[det].lumi_coord
-
-  def empty(self):
-    for det in self.det_map:
-      if (any(self.det_map[det].vis_coord) or any(self.det_map[det].vis_M)
-          or any(self.det_map[det].lumi_coord)):
-        return False
-    return True
-
-  @staticmethod
-  def roundz(rawz):
-    return round(rawz, 1)
+
+   jsonmap = json.loads(open(filename, 'r').read())
+   self.filename = filename
+   # only load the board if the file contains the required fields
+   if 'type' in jsonmap and 'description' in jsonmap and 'detectors' in jsonmap and len(jsonmap['detectors'].keys()) > 0:
+       jsonmap = json.loads(open(filename, 'r').read())
+       self.type = jsonmap['type']
+       self.description = jsonmap['description']
+       self.id = jsonmap['id'] if 'id' in jsonmap else -1
+       self.calib_routines = jsonmap['calib_routines'] if 'calib_routines' in jsonmap else []
+       self.conditions = jsonmap['conditions'] if 'conditions' in jsonmap else {}
+
+       for detid in range(1, len(jsonmap['detectors']+1)):
```

Comment:
Parsing typo: `len(jsonmap['detector']) +1 `

Reply:
it is supposed to be 'detectors' I believe, isn't it?

Reply:
Ah, I was concerned about where the `len` parenthesis encapsulates.